context/rootfs/scripts/init-registry.sh

```diff
@@ -16,70 +16,118 @@
+}
+
+CONTAINER_RUNTIME=docker
+if ! utils_command_exists docker; then
+  if containerd --version; then
+    CONTAINER_RUNTIME=containerd
+  else
+    echo "either docker and containerd not found, please check" >&2
+    exit 1
+  fi
+fi
+
 container=sealer-registry
 rootfs=$(dirname "$(pwd)")
 config="$rootfs/etc/registry_config.yml"
 htpasswd="$rootfs/etc/registry_htpasswd"
 certs_dir="$rootfs/certs"
+image_dir="$rootfs/images"
 
 mkdir -p "$VOLUME" || true
 
-imageServer="nerdctl"
-if [[ $(ls ../cri/docker*.tar.gz) ]]; then
-    imageServer="docker"
-fi
+# shellcheck disable=SC2068
+runtimeRun() {
+  if [ "$CONTAINER_RUNTIME" == "containerd" ]; then
+    nerdctl container run $@
+  else
+    docker run $@
+  fi
+}
+
+# shellcheck disable=SC2068
+runtimeStart() {
+  if [ "$CONTAINER_RUNTIME" == "containerd" ]; then
+    nerdctl start $@
+  else
+    docker start $@
+  fi
+}
+
+# shellcheck disable=SC2068
+runtimeInspect() {
+  if [ "$CONTAINER_RUNTIME" == "containerd" ]; then
+    nerdctl container inspect $@
+  else
+    docker inspect $@
+  fi
+}
+
+# shellcheck disable=SC2068
+runtimeGetContainerStatus() {
+  if [ "$CONTAINER_RUNTIME" == "containerd" ]; then
+    nerdctl container inspect $@ | grep '"Status"' | awk '{print $2}' | tr -d ','
+  else
+    docker inspect --format '{{json .State.Status}}' $@
+  fi
+}
 
-start_registry() {
+startRegistry() {
   n=1
   while ((n <= 3)); do
     echo "attempt to start registry"
-    $imageServer start $container; break || ((n++))
+    runtimeStart $container && break
+    ((n++))
     sleep 3
   done
 }
 
-docker_check_registry() {
-  n=1
-  while (( n <= 3 ))
-  do
-      registry_status=$(docker inspect --format '{{json .State.Status}}' sealer-registry)
-      if [[ "$registry_status" == \"running\" ]]; then
-          break
-      fi
-      if [[ $n -eq 3 ]]; then
-         echo "sealer-registry is not running, status: $registry_status"
-         exit 1
-      fi
-      (( n++ ))
-      sleep 3
+load_images() {
+  for image in "$image_dir"/*; do
+    if [ ! -f "${image}" ]; then
+      continue
+    fi
+    if [ "$CONTAINER_RUNTIME" == "containerd" ]; then
+      ctr image import "${image}"
+    else
+      docker load -q -i "${image}"
+    fi
   done
 }
 
-nerdctl_check_registry() {
+check_registry() {
   n=1
-  while (( n <= 3 ))
-  do
-    if nerdctl inspect sealer-registry | grep -q '"Status": "running"'; then
-        break
-    fi
-    if [[ $n -eq 3 ]]; then
-      ## TODO: Add registry_status of sealer-registry container
-      echo "sealer-registry is not running, use 'nerdctl inspect sealer-registry' command to show status"
-      exit 1
-    fi
-    (( n++ ))
+  while ((n <= 3)); do
+    registry_status=$(runtimeGetContainerStatus sealer-registry)
+    [[ "$registry_status" == \"running\" ]] && break
+    ((n++))
     sleep 3
   done
+  if [[ "$registry_status" != \"running\" ]]; then
+    echo "sealer-registry is not running, status: $registry_status"
+    return 1
```

Comment:
exit 1, rather than return 1.

And make the error info to output into stderr rather stdout.